src/server/page_server.hpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2008 Search Solution Corporation
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#ifndef _PAGE_SERVER_HPP_
+#define _PAGE_SERVER_HPP_
+
+#include "ats_ps_request.hpp"
+#include "request_client_server.hpp"
+
+// forward declaration
+namespace cubpacking
+{
+  class unpacker;
+}
+
+class page_server

```

Comment:
would be good to mark all classes that are not supposed to be derived from - or at least which are not supposed to be derived from for now - with `final`

Reply:
Not sure.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c139-use-final-on-classes-sparingly

Reply:
You have a point. I was not aware.